.github/workflows/check-strava-api.yml

```diff
@@ -0,0 +1,29 @@
+name: Check Strava API
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+
+jobs:
+  update-model:
+    name: Update Model
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Fetch Latest API
+        uses: stravalib/strava_swagger2pydantic@v1
+        with:
+          model_file: 'stravalib/strava_model.py'
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          add-paths: |
+            stravalib
+          commit-message: Strava API Change
+          branch: api-change
```

Comment:
@jsamoocha can you help me understand the workflow here. i see a pydantic model branch but this will pull against api-change

Reply:
This workflow has three steps:

1. Check out the stravalib repo
2. Generate the Pydantic model from Strava's published API, which _may_ update the file `stravalib/strava_model.py`. This module contains the classes of the Pydantic model.
3. Create a temporary branch `api_change`, and in case of any git diffs in the path `stravalib`, create a PR with the provided commit message, title, etc.

So the `branch` key is just a parameter for the [create-pull-request action](https://github.com/marketplace/actions/create-pull-request). I agree the authors of that action could've prevented some confusion by naming that key `tmp-branch-name` or whatever else.

This workflow will run (AFAIK) only on the master branch when scheduled and can run on any branch when triggered manually.

I'm new to GitHub Actions and Workflows, so please feel free to ask more in case I don't make sense.

Reply:
ok i think i follow now! let's merge as it's not going to modify our main code base at this point yet. 

i will play with the script that you wrote to actually create the API when i have a bit more time - more for my learning than as a test :)

Reply:
I tested it on a private toy repo. I just gave you access to it to save some boilerplate code/time. Feel free to modify whatever you want for testing/learning purposes.